athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/HbaseTableNameUtils.java

```diff
@@ -0,0 +1,152 @@
+ * a case insensitive search over them. It will first try to do a targeted get to reduce the penalty for
+ * tables which don't have capitalization.
+ * 
+ * Modeled off of DynamoDBTableResolver.java
+ *
+ * TODO add caching
+ */
+public final class HbaseTableNameUtils
+{
+    //The HBase namespce qualifier character which commonly separates namespaces and column families from tables and columns.
+    protected static final String NAMESPACE_QUALIFIER = ":";
+    private static final Logger logger = LoggerFactory.getLogger(HbaseTableNameUtils.class);
+
+    private HbaseTableNameUtils() {}
+    
+    /**
+     * Helper which goes from a schema and table name to an HBase table name string
+     * @param schema a schema name
+     * @param table the name of the table
+     * @return
+     */
+   public static String getQualifiedTableName(String schema, String table)
+   {
+       return schema + NAMESPACE_QUALIFIER + table;
+   }
+
+    /**
+     * Helper which goes from an Athena Federation SDK TableName to an HBase table name string.
+     *
+     * @param tableName An Athena Federation SDK TableName.
+     * @return The corresponding HBase table name string.
+     */
+    public static String getQualifiedTableName(TableName tableName)
+    {
+        return getQualifiedTableName(tableName.getSchemaName(), tableName.getTableName());
+    }
+
+    /**
+     * Helper which goes from a schema and table name to an HBase TableName
+     * @param schema the schema name
+     * @param table the name of the table
+     * @return The corresponding HBase TableName
+     */
+    public static org.apache.hadoop.hbase.TableName getQualifiedTable(String schema, String table)
+    {
+        return org.apache.hadoop.hbase.TableName.valueOf(getQualifiedTableName(schema, table));
+    }
+
+    /**
+     * Helper which goes from an Athena Federation SDK TableName to an HBase TableName.
+     *
+     * @param tableName An Athena Federation SDK TableName.
+     * @return The corresponding HBase TableName.
+     */
+    public static org.apache.hadoop.hbase.TableName getQualifiedTable(TableName tableName)
+    {
+        return org.apache.hadoop.hbase.TableName.valueOf(getQualifiedTableName(tableName));
+    }
+
+    /**
+     * Gets the hbase table name from Athena table name. This is to allow athena to query uppercase table names
+     * (since athena does not support them). If an hbase table name is found with the athena table name, it is returned.
+     * Otherwise, tryCaseInsensitiveSearch is used to find the corresponding hbase table.
+     *
+     * @param tableName the case insensitive table name
+     * @return the hbase table name
+     */
+    public static org.apache.hadoop.hbase.TableName getHbaseTableName(HBaseConnection conn, TableName athTableName)
+            throws IOException
+    {
+        if (conn.tableExists(getQualifiedTable(athTableName))) {
```

Comment:
Double check, we lower case the athTableName here. If Hbase has Uppercase, conn.getTableExists will able to find that table?

Reply:
This is a check to see if the table we are trying to get is already lowercase. If it is (tableExists), then we can skip the rest of the logic. Otherwise, we run the caseInsensitiveSearch.